pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -79,15 +79,18 @@ func (e *timeSeriesQuery) processQuery(q *Query, ms *es.MultiSearchRequestBuilde
 	}
 
 	if len(q.BucketAggs) == 0 {
-		if len(q.Metrics) == 0 || q.Metrics[0].Type != "raw_document" {
+		if len(q.Metrics) == 0 || !(q.Metrics[0].Type == "raw_document" || q.Metrics[0].Type == "raw_data") {
 			result.Responses[q.RefID] = backend.DataResponse{
 				Error: fmt.Errorf("invalid query, missing metrics and aggregations"),
 			}
 			return nil
 		}
 		metric := q.Metrics[0]
 		b.Size(metric.Settings.Get("size").MustInt(500))
+		// Should this be time field (data source or query?)
```

Comment:
@gabor Just a sanity check - Am I correct to expect that at all places here, where we use `@timestamp`, we should use configured `time field`

Reply:
hmm... for example, when you use `date_histogram`, then you can specify the time-field you want to use for it.
i assume we default to the datasource-configured-time-field, but not 100% sure.

for things like `logs` and `raw_data`, i assume we take the datasource-configured-time-field and use that to construct the elastic-query.

Reply:
i did some digging, and for date_histogram we do:
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/QueryBuilder.ts#L97

so we take whatever is in the aggregation, and fallback to the datasource-configured one.

for other places, like this one (`raw_data`), i agree, we should use the datasource-configured field.

we do not allow an unset datasource-time-field ( https://github.com/grafana/grafana/blob/main/pkg/tsdb/elasticsearch/elasticsearch.go#L102 ), so we can rely on it always being available.